### ISSUE TITLE
Launchers not adding to Steam

### ISSUE THREAD

**Issue by Betablockr** [moraroy/NonSteamLaunchers-On-Steam-Deck#269]
I'm using the latest NonSteamLauncher version and I am unable to get any of the launchers to add to Steam. They seemingly have an install path however.

Also the Xbox Game Pass app refuses to install regardless on if I have Chrome, Chromium or Edge installed so I am unable to manually link this. 

I did first try installing on Steam OS Beta as I use the new family option. I did think after a recent RMA think they may have reimaged it wrong so did a re image and stayed on the stable OS and tried again. I got the same result. 

I also tried the Decky Loader option which on one run did successfully add Epic, GOG and EA but only GOG run.

Manually adding the install path to Steam also didn't work. 

I'm a bit lost on what to do?

**Comment by Piiikachu** [comment]
Same problem. No launchers add to steam. I wonder if it is because there are two accounts in my steam. Hope someone can fix it.

**Comment by Betablockr** [comment]
I saw on a previous post that someone said to manually ad a non steam game OR when in game mode install Google Chrome when promted. I missed this prompt to manually add chrome to the Steam games. Now GOG, Epic, EA and Rockstar work BUT xbox cloud app refuses to appear. Very frustrating

**Comment by gertis99** [comment]
Hey. I had the same problem, but I fixed this checking the box where says spearate app ids, and checking the box for xbox game pass. I hope this helps you.

**Comment by Starlynk1** [comment]
What I've found is that if you try to do a fresh start, it doesn't remove the app id folders and then fails to accept a new install of that app. If you plan on doing a fresh start, then I recommend going into the following folder and remove any uninstalled non steam items.

Home/.local/share/Steam/steamapps/compatdata

Look into each subfolder until you find the app it is referenced to. Usually in the Program Files (x86) folder.

**Comment by Betablockr** [comment]
> What I've found is that if you try to do a fresh start, it doesn't remove the app id folders and then fails to accept a new install of that app. If you plan on doing a fresh start, then I recommend going into the following folder and remove any uninstalled non steam items.
> 
> Home/.local/share/Steam/steamapps/compatdata
> 
> Look into each subfolder until you find the app it is referenced to. Usually in the Program Files (x86) folder.

So I did try a factory reset on my Steam deck and.... nothing. But then I did a re-image (I also use the steam client beta) and this time around it worked, so maybe it was a dodgy re-inage after a recent valve repair on my headphone port. I used the testing version of nonsteamlauchers via the deckyloader plug in which worked perfectly. However I had to first do the "install chrome and add to home screen" feature to get the xbox game pass to work properly, a manual install of Chrome didn't seem to work.

Unsure if this is a resolved issue because it might still be something you can replicate easy. Or likely see happen again.

**Comment by moraroy** [comment]
@Betablockr the problems youre describing some of them have actually been fixed. Im waiting on the decky loader team to accept those changes! but yes im glad the plugin is working nice for you. keep on the look out for more changes and updates hopefully will be able to move to main soon!

**Comment by livejamie** [comment]
I had similar problems, I had to restart steam to get a bunch of loaders to work with the decky version.

**Comment by moraroy** [comment]
@livejamie yeah that one is out of my control

**Comment by S0R0SH** [comment]
I have same problems, NonSteam didn't add launchers to Steam. Please Fix it Dear Dev 😢😢😢😢👈👈👈👈

**Comment by moraroy** [comment]
@S0R0SH are you using the plugin version or the desktop?

**Comment by S0R0SH** [comment]
> @S0R0SH are you using the plugin version or the desktop?

Desktop and try 5 times, with and without proxy, Even after installing it, it removed the DeckyLoader 😂

**Comment by moraroy** [comment]
@S0R0SH I don't know what that means, can you provide your log in the downloads folder?

**Comment by S0R0SH** [comment]
> @S0R0SH I don't know what that means, can you provide your log in the downloads folder?

Sorry, I will explain again. I have done the installation several times. At the end of the installation, no launcher was added to my Steam. I had installed the YouTube option only once and it installed correctly.

I want to reinstall the program right now. Then I will send you the log file. thanks for your help 🙏🙏🙏

**Comment by ldebien** [comment]
Just wanted to add on this issue.

I just installed NSL 3.9.0 (from sources code realease with the .desktop script) for the first time on my SD desktop. I ran the install for Epic Games (which failed due to corrupted files), GOG Galaxy (which opened the gog installer wizard) and Ubisoft Connect. After it did all this, Steam restarted in background, which I believe is fine. But none of the three launchers were present in steam. So I launched again the NSL script (still from desktop) and it appeared that both GOG Galaxy and Ubisoft had a path. I did a Fresh Start and tried again with only Ubisoft Connect. The result was the same (i.e. Installing seems to go OK but nothing shows in my Steam Library). I restarted the Steam Deck but still nothing.

I did a second test though. I did another Fresh Start then updated Decky Loader and went to Gaming Mode to install the NSL plugin. I tried installing Ubisoft Connect and got immediately a "Launcher Installed" notification. It was still not present in my Steam Library though. I tried to do the same install with EA app with the same result.

Some notes : I've got Proton-QT installed since I used GE-Proton prior to installing NSL. I tried to install myself Ubisoft connect a year ago or so, it didn't work well (was asked my password each time I started a game). The install is still present on my SD. Maybe it's what caused an issue?

To me, NSL is unusable right now.

**Comment by S0R0SH** [comment]
> Just wanted to add on this issue.
> 
> I just installed NSL 3.9.0 (from sources code realease with the .desktop script) for the first time on my SD desktop. I ran the install for Epic Games (which failed due to corrupted files), GOG Galaxy (which opened the gog installer wizard) and Ubisoft Connect. After it did all this, Steam restarted in background, which I believe is fine. But none of the three launchers were present in steam. So I launched again the NSL script (still from desktop) and it appeared that both GOG Galaxy and Ubisoft had a path. I did a Fresh Start and tried again with only Ubisoft Connect. The result was the same (i.e. Installing seems to go OK but nothing shows in my Steam Library). I restarted the Steam Deck but still nothing.
> 
> I did a second test though. I did another Fresh Start then updated Decky Loader and went to Gaming Mode to install the NSL plugin. I tried installing Ubisoft Connect and got immediately a "Launcher Installed" notification. It was still not present in my Steam Library though. I tried to do the same install with EA app with the same result.
> 
> Some notes : I've got Proton-QT installed since I used GE-Proton prior to installing NSL. I tried to install myself Ubisoft connect a year ago or so, it didn't work well (was asked my password each time I started a game). The install is still present on my SD. Maybe it's what caused an issue?

I went through exactly the same steps and got the same results, NSL 3.9.0 NSL 3.9.0 didn't work again !!!!!!!! ☹☹☹

**Comment by moraroy** [comment]
@S0R0SH can you send me your log in the downloads folder?

**Comment by z-eh** [comment]
problem still exists...

**Comment by moraroy** [comment]
@z-eh launchers arent adding?

**Comment by z-eh** [comment]
> [@z-eh](https://github.com/z-eh) launchers arent adding?

Yes, I tried everything but the shortcuts not created in the steam library... :(

**Comment by moraroy** [comment]
@z-eh are you on Desktop or decky plugin?

**Comment by z-eh** [comment]
tried both, nothing

**Comment by moraroy** [comment]
@z-eh ok, you need to look at the logs for decky plugin go to /home/deck/homebrew/logs/nonsteamlaunchers upload the most recent one with you running a scan so we can see what's going on

**Comment by Arianneeee** [comment]
Currently having this problem right now after I factory reset my steam deck. It was fine before the factory reset.

**Comment by 00Zazhary00** [comment]
> [@z-eh](https://github.com/z-eh) ok, you need to look at the logs for decky plugin go to /home/deck/homebrew/logs/nonsteamlaunchers upload the most recent one with you running a scan so we can see what's going on

it says "required JSON files missing or empty" "Skipping Scan" and "Failed to Start NonSteamLaunchers!"

**Comment by MillionIncredible** [comment]
I'm having the same problem. I'm using the desktop version. When I try to install Ubi Connect, for example, it only adds Moonlight again (as I already have that in Steam), but not Ubi Connect.

**Comment by IlBeduino8055** [comment]
hello everyone, I only discovered NSL today and it's cool, I compliment the developer. except I'm having a problem, I'm trying to download Ubisoft Connect on my Steam Deck, using NSL plugin with Decky Loader but when I try to install Ubisoft Connect it tells me "successfully installed" but I don't see any shortcuts as a result I can't find it. I have done several tests on clean installations and repeated uninstallations, I also tried using the Desktop application but nothing to do. I thought the problem was the latest update to the Steam Deck, could it be? how can I do? I leave the Log attached. 

[NonSteamLaunchers-install.log](https://github.com/user-attachments/files/26649613/NonSteamLaunchers-install.log)

**Comment by moraroy** [comment]
@llbeduino8055 fixed please update

**Comment by moraroy** [comment]
@MillionIncredible should be fixed as of now please update
